cmd/kubeadm/app/cmd/phases/kubelet.go

```diff
@@ -222,6 +232,45 @@ func NewCmdKubeletConfigUpload() *cobra.Command {
 	return cmd
 }
 
+// NewCmdKubeletAnnotateCRI calls cobra.Command for annotating the node with the given crisocket
+func NewCmdKubeletAnnotateCRI() *cobra.Command {
+	cfg := &kubeadmapiv1alpha3.InitConfiguration{}
+	var cfgPath string
+	kubeConfigFile := constants.GetAdminKubeConfigPath()
+
+	cmd := &cobra.Command{
+		Use:     "annotate-cri",
+		Short:   "annotates the node with the given crisocket",
+		Long:    kubeletConfigAnnotateCRILongDesc,
+		Example: kubeletConfigAnnotateCRIExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			if len(cfgPath) == 0 {
+				kubeadmutil.CheckErr(fmt.Errorf("The --config argument is required"))
+			}
+
+			// KubernetesVersion is not used, but we set it explicitly to avoid the lookup
+			// of the version from the internet when executing ConfigFileAndDefaultsToInternalConfig
+			err := SetKubernetesVersion(nil, cfg)
```

Comment:
sorry, i was so tired last night that i might have missed this.

i'm pretty sure that ConfigFileAndDefaultsToInternalConfig wil end up calling this:
https://github.com/kubernetes/kubernetes/blob/ed958b7d79ee7eee0d71952e71475fe667c77909/cmd/kubeadm/app/util/version.go#L62

which handles the look up from the internet. it tries it, but then it falls back to the client version in air-gapped cases.

SetKubernetesVersion() on the hand falls back to the client version directly:
https://github.com/kubernetes/kubernetes/blob/7f23a743e8c23ac6489340bbb34fa6f1d392db9d/cmd/kubeadm/app/phases/upgrade/versiongetter.go#L74

i guess it's a question if want to always first fall back to the client version for this phase?

Reply:
IMO Let's gets this into beta-2 an then eventually iterate for the air gapped/offline cases

Reply:
ok, sounds good.
i guess my point was that the call to `SetKubernetesVersion` might be redundant, but i haven't verified.